### PR TITLE
Add ability to view puppet classes and types for individual hosts/nodes

### LINKED ISSUE #53: Ability to view node classes and types
We should add the ability to view information about a node/host's puppet classes and types.

The use should be able to view this for individual nodes listed under `/puppetdb_foreman/nodes` and for individual host's already in foreman.

### PR DESCRIPTION
**Pull request by leewaa**
Fixes #53.

### CODE REVIEW

File: app/controllers/puppetdb_foreman/nodes_controller.rb [hunk 1]
```diff
@@ -12,6 +12,18 @@ def index
       @unknown_hosts = @puppetdb_hosts - @foreman_hosts
     end
 
+    def show
+      @params = params
```
Comment (timogoebel) [inline app/controllers/puppetdb_foreman/nodes_controller.rb]: Where is this used?

File: app/views/puppetdb_foreman/nodes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<% title _('PuppetDB Node') %>
+
+<%= host_breadcrumb %>
```
Comment (timogoebel) [inline app/views/puppetdb_foreman/nodes/show.html.erb]: This needs foreman >= 1.18. Can you please either add a guard (`respond_to?(: host_breadcrumb)`) for this or bump the required foreman version in `lib/puppetdb_foreman/engine.rb`?
Also, you can omit the `=` here. `<% host_breadcrumb` should be enough.

File: app/views/puppetdb_foreman/nodes/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                  <tr>
+                    <th class="col-md-5"><%= _('Name') %></th>
+                  </tr>
+                </thead>
+                <tbody>
+                  <% @classes.each do |class_name| %>
+                    <tr><td><%= class_name %></td></tr>
+                  <% end %>
+                </tbody>
+	            </table>
```
Comment (timogoebel) [inline app/views/puppetdb_foreman/nodes/show.html.erb]: Nit: indentation is off

File: app/views/puppetdb_foreman/nodes/index.html.erb [hunk 4]
```diff
@@ -51,6 +51,7 @@
                       <td class="ellipsis"><%= node %></td>
                       <td>
                         <%= action_buttons(
+                          (display_link_if_authorized(_("Preview"), hash_for_puppetdb_foreman_node_path(:id => node), :method => :get)),
```
[the remark this answers is not in the document]
Comment (leewaa) [reply app/views/puppetdb_foreman/nodes/index.html.erb]: Not sure how this `display_link_if_authorized` stuff works or what exactly it is expecting. `Preview` is appropriate as the link label however.

**Comment by leewaa** [comment]
I will change the breadcrumb bar so that it works for nodes, not hosts :D :D

File: app/views/puppetdb_foreman/nodes/index.html.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (timogoebel) [reply app/views/puppetdb_foreman/nodes/index.html.erb]: I was just wondering about the label...

[the remark this answers is not in the document]
Comment (leewaa) [reply app/views/puppetdb_foreman/nodes/index.html.erb]: I see. I think Preview is good, don't you think ?

Its basically a preview of the node to determine whether or not to import it into foreman.

[the remark this answers is not in the document]
Comment (timogoebel) [reply app/views/puppetdb_foreman/nodes/index.html.erb]: Yeah, or you could just use the label "Puppet classes" or "Classes". That's even easier.

**Comment by leewaa** [comment]
@timogoebel I ended up leaving the breadcrumb switcher out for now, it needs some work to be able to work with non active-record objects, and some more thought needs to be put into the current API also. Foreman does allot fo work in this respect and I don't want add all that dependancy to this Plugin.

For now the breadcrumb items are defined manually as the default does not work out of the box. I could pack it away into a helper, but I like the more explicit approach in the view.

File: app/helpers/concerns/puppetdb_foreman/hosts_helper_extensions.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    module Overrides
+      def show_appropriate_host_buttons(host)
+        buttons = super
+
+        buttons << link_to_if_authorized(
+          _('Puppet Classes'),
+          hash_for_puppetdb_foreman_node_path(:id => host.name),
+          :title => _('Browse host puppet classes'),
+          :class => 'btn btn-default'
+        )
```
Comment (timogoebel) [inline app/helpers/concerns/puppetdb_foreman/hosts_helper_extensions.rb]: Can we add a guard for this to only display the button if the host has puppet enabled? Maybe something like `if host.puppet_proxy_id?`. It's not perfect, but it saves a query to PuppetDB to check if the host is present there.

Comment (leewaa) [reply app/helpers/concerns/puppetdb_foreman/hosts_helper_extensions.rb]: The guard you suggested works perfectly :)

**Comment by timogoebel** [comment]
@leewaa: Can you please check out the failing test?

```
AccessPermissionsTest.test_0742_route puppetdb_foreman/nodes/show should have a permission that grants access
```

I believe defining a permission for the route in engine.rb should be enough.

**Comment by timogoebel** [comment]
[test puppetdb_foreman]

**Comment by timogoebel** [comment]
Thanks, @leewaa.

**Pull request merged**